core/tests/protractor_utils/ProfilePage.js

```diff
@@ -89,8 +98,12 @@ var ProfilePage = function() {
     var explorationsCardByName = await allExplorationCardElements.filter(
       async function(card) {
         var cardTitle = card.element(cardTitleCss);
-        var title = await cardTitle.getText();
-        return title === explorationName;
+        var title = await allExplorationCardElements.map(async function(cardTitle) {
+          await waitFor.visibilityOf(
+            cardTitle,
+            'cardTitle is not present or taking Too Long to Dispaly');
+          expect(await cardTitle.getText()).toMatch(explorationName);
+        });
```

Comment:
You didn't need to make such a big change here. Just add a waitFor call before ` var title = await cardTitle.getText();`. Otherwise, the code was fine before